python/sparkdl/utils/generate_app_models.py

```diff
@@ -0,0 +1,156 @@
+#!/bin/python
+
+# Copyright 2017 Databricks, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# NOTE: Run from project's root directory
+# Generate TF graphs for DeepImageFeaturizer.
+#
+# Takes keras models in sparkdl.transformers.keras_applications and prepends reshaping from ImageSchema
+# and model specific preprocessing.
+# Produces tensor flow model files and a scala file containing scala wrappers for all the models.
+#
+# Input: sparkdl.transformers.keras_aplications.KERAS_APPLICATION_MODELS
+#
+# Output: model *.pb files in the working directory.
+#
+from base64 import b64encode
+from hashlib import sha256
+import os
+
+import tensorflow as tf
+import keras.backend as K
+
+from sparkdl.graph import utils as tfx
+from sparkdl.transformers import *
+from sparkdl.transformers.keras_applications import *
+from sparkdl.transformers.named_image import *
+
+scala_template = """
+  private[sparkdl] object %(name)s extends NamedImageModel {
+    override val name = "%(name)s"
+    override val height = %(height)d
+    override val width = %(width)d
+    override val graphInputNode = name + "_input"
+    override val graphOutputNode = name + "_sparkdl_output__"
+
+    override def graph: GraphDef = ModelFetcher.getFromWeb(
+        "https://s3-us-west-2.amazonaws.com/spark-deep-learning-models/sparkdl-%(name)s_v%(version)d.pb",
+        fileName = "sparkdl-inceptionV3_v%(version)d.pb",
+        base64Hash = "%(base64)s"
+    )
+  }
+"""
+
+
+def gen_model(name, model, model_file, version=1, featurize=True):
+    g = tf.Graph()
```

Comment:
Simpler to read & understand that there is no scope overlap:
```
g = tf.Graph()
with tf.Session(graph=g) as sess:
  ...
  gdef =  tfx.strip_and_freeze_until([outTensor], sess.graph, sess, return_graph=False)
g2 = tf.Graph()
with tf.Session(graph=g2) as sess:    # i believe `with g2.as_default():` instead of initializing a session would also work here but I'd do it outside of the previous session for clarity.
  tf.import_graph_def(gdef, name='')
  filename = "sparkdl-%s_v%d.pb" % (name, version)
  print 'writing out ', filename
  tf.train.write_graph(g2.as_graph_def(), logdir="./", name=filename, as_text=False)
# I'd return here and deal with the scala file writing elsewhere, but if you want to put anything more, you can do it outside the second session.
```

Reply:
Yeah that's way cleaner, thanks.